core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -86,18 +84,17 @@ private[spark] class EventLoggingListener(
 
   private def initEventLog(): Unit = {
     val metadata = SparkListenerLogStart(SPARK_VERSION)
-    val eventJson = JsonProtocol.logStartToJson(metadata)
-    val metadataJson = compact(eventJson)
-    logWriter.writeEvent(metadataJson, flushLogger = true)
+    val eventJson = JsonProtocol.sparkEventToJsonString(metadata)
+    logWriter.writeEvent(eventJson, flushLogger = true)
```

Comment:
Not related to this PR: does it mean we generate a big JSON string and write it to files? Is it possible to do it in a streaming fashion?

Reply:
Yes, the current code needs to materialize each event as a single JSON string. In principle we could do the write in a streaming fashion by keeping a handle to an output stream. Streaming writes could make it slightly trickier to recover from individual event writing failures, though (e.g. try to serialize but bail out and drop an event if the write fails).